pkg/dataobj/internal/sections/logs/logs.go

```diff
@@ -21,10 +21,11 @@ import (
 
 // A Record is an individual log record within the logs section.
 type Record struct {
-	StreamID  int64
-	Timestamp time.Time
-	Metadata  labels.Labels
-	Line      []byte
+	StreamID    int64
+	Timestamp   time.Time
+	Metadata    labels.Labels
+	Line        []byte
+	MdValueCaps []int
```

Comment:
I had to do this in a few places. Is there a better way to return buffers to their original capacity when reusing them? 

Every time something turns into a string, we lose the original capacity of the underlying slice/array so I needed to store these separately to resize them again later.

Reply:
Can you point me at where we lose the original capacity of the slice? It's not obvious to me from reviewing the PR.